### ISSUE TITLE
bad interpreter Init Debian

### ISSUE THREAD

**Issue by jniltinho** [gogits/gogs#1488]
-bash: /etc/init.d/gogs: /bin/sh^M: bad interpreter: No such file or directory

sed -i -e 's/\r$//' /etc/init.d/gogs

**Comment by jniltinho** [comment]
Update Script Auto Install Gogs on Ubuntu 14.04
https://gist.github.com/jniltinho/280573e10206e31ab1c7

**Comment by tboerger** [comment]
You know about the debian/ubuntu packages of gogs? :)

**Comment by jniltinho** [comment]
Link: http://gogs.dn.qbox.me/gogs_v0.6.3_linux_amd64.zip

**Comment by Unknwon** [comment]
Sorry I don't understand the issue, where is the `/etc/init.d/gogs` in Gogs repository?

**Comment by jniltinho** [comment]
Error Bash file.

http://askubuntu.com/questions/304999/not-able-to-execute-a-sh-file-bin-bashm-bad-interpreter
http://tombevers.me/2012/05/17/how-to-fix-a-binshm-bad-interpreter-error/

File: gogs/scripts/init/debian/gogs

**Comment by Unknwon** [comment]
Sounds like a very good PR :stuck_out_tongue_closed_eyes:

**Comment by Em-AK** [comment]
I had the same issue :

1. download the latest gogs zip archive
2. extract it with unzip
3. copy `gogs/scripts/init/debian/gogs` into my `/etc/init.d/gogs
4. `/etc/init.d/gogs start` returns a `bash: /etc/init.d/gogs: /bin/sh^M: bad interpreter: No such file or directory`

After checking the file format it is not "unix style"
```
git@multigems:~$ file gogs/scripts/init/debian/gogs  
gogs/scripts/init/debian/gogs: POSIX shell script, ASCII text executable, **with CRLF line terminators**
```
I fixed it with `dos2unix /etc/init.d/gogs` 

I just wonder why the file format is wrong? Is it the `unzip` command that creates the file with the wrong format ?

**Comment by Unknwon** [comment]
Hi all, I've checked that the file itself is using Unix style line endings.

However, all binary(zip) files are packed under Windows, so it is possible that Zip software under Windows I use somehow converts them into Windows style line endings.

**Comment by Unknwon** [comment]
This is should be fixed by 0.6.9 as I use linux to pack ZIP archive.
